coderd/oauth2_validation.go

```diff
@@ -0,0 +1,361 @@
+package coderd
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"net/url"
+	"slices"
+	"strings"
+
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/codersdk"
+)
+
+// RFC 7591 validation functions for Dynamic Client Registration
+
+// validateClientRegistrationRequest validates the entire registration request
+func validateClientRegistrationRequest(req codersdk.OAuth2ClientRegistrationRequest) error {
```

Comment:
nit: this could be a `.Valid()` method on the codersdk object. Same for below.

Reply:
refactored along with all the other functions.